File: webapp/frontend/views/uix/components/circular_icon_button/circular_icon_button.py
import os
import random
import tornado.web

from typing import List


class CircularIconButton(tornado.web.UIModule):
    
    def css_files(self):
        return [os.path.join("css", "circular_icon_button.css")]

    def render(self, icon_path:str, key:str, alt:str = "", title:str = ""):
        return self.render_string(
            "circular_icon_button.html",
            icon_path=icon_path,
            key=key, 
            alt=alt,
            title = title)

File: webapp/frontend/views/uix/components/circular_icon_button/circular_icon_button.ts
class GenerateCircularIconButton
{
    #icon:string = ''
    #alt:string = ''
    #title:string = ''
    #key:string = ''
    constructor(icon_path:string, key:string,  alt:string='', title:string='')
    {
        this.#icon = icon_path
        this.#key = key
        this.#alt = alt
        this.#title = title
    }

    render()
    {
      return `\
        <button type="menu" class="circular-icon-button" key='${this.#key}'>
            <img src="${this.#icon}" alt="${this.#alt}" title="${this.#title}"/>
        </button>`  
    }
}


export default GenerateCircularIconButton